code/modules/mining/ore_detector.dm

```diff
@@ -168,6 +178,22 @@
 /obj/item/ore_detector/on_give()
 	deactivate()
 
+// horrendous hack, but it's an engine limitation
+// the way images work is that it's only show to a client when you add it via client.images += I
+// the problem with that is it needs a loc to attach to, and if you attach it to a turf, the image is only visible if that turf is visible
+// meaning that you can't see it through walls, which is the whole point of the detector
+// so, what we do instead, is spawn this anchor beneath the player and attach all the images to the anchor, then pixel shift them to the turf it needs to render over
+// the reason we're making the anchor instead of just making the loc our turf, is that clicking on an image passes the click through to whatever it's attached to, it works like an overlay
+// so the right click menu gets messed up, and clicking on the ore blip means you actually click beneath yourself, which can be disastrous
+// so, by having an anchor with MOUSE_OPACITY_TRANSPARENT, we can circumvent ALL those issues
```

Comment:
Shouldn't this be done at the HUD level then? It seems this is supposed to act more like an HUD element than an ingame object, so it might aswell be one?

Reply:
i tried using the HUD, but it was _incredibly_ disorientating when the blips moved along with you but the turfs stayed static

Reply:
I don't mean the HUD as in the HUD itself, but the layer of the HUD, above the occlusion plane, that that way shouldn't be occluded I think (?)

Reply:
i messed around with planes and layers a lot to no success, the issue isn't really the darkness more than it is the turfs simply not being rendered at all, due to how images work as overlays in the backend